### ISSUE TITLE
 Update Lucene to use Japanese

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/airsonic#4]
## Problem description

Because lucene version is old, Japanese can not be used.

Strictly speaking, effective Japanese morpheme analysis can not be performed because Japanese libraries necessary for additional processing can not be used.

I can not sort using Japanese reading.
Also, since ngram is effectively a 1-gram, mis-retrieval becomes very numerous.

## System information

Airsonic : 10.2.0-SNAPSHOT
lucene-core : 3.0.3

## Additional notes

I recognize that Lucene's update is mainly a sensitive issue in terms of performance.

 - [Upgrade Lucene to improve Unicode handling](https://github.com/airsonic/airsonic/issues/135)
 - [Libresonic v6.3 Planning](https://github.com/airsonic/airsonic/issues/374)

Since Japanese morpheme analysis is special, development is slower than multilingual.
Specifically, Japanese processing library requires at least Lucene 4.x or later (· _ ·,)
It is better if it is new.

[Jpsonic](https://github.com/tesshucom/jpsonic) has already built in lucene 7.4 and has a track record.
It will be upgraded to lucene 7.6 soon.
(It does not measure large-scale performance, but it is practically used)

Extract lucene updates and autovacuum update difference (excluding Japanese processing) and create branch.
(In the case of pure performance measurement of lucene, since a phase that eliminates processing related to Japanese is necessary)

**Comment by tesshucom** [comment]
I updated lucene to already experienced 7.4.
(Update later to 7.6)
I tried as much processing as possible to make it easier for Diff to do but it is not exactly the same.
I think most are due to grammar and architect differences between 3.x and 7.x.

Compiling with travis has passed the test.
However, I analyze the generated index with luke and make a manual check as to whether the equivalent is created.
